File: android/src/main/java/com/easemob/ext_sdk/common/ExtSdkDispatch.java
package com.easemob.ext_sdk.common;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class ExtSdkDispatch implements ExtSdkApi{

    private static class SingleHolder {
        static ExtSdkDispatch instance = new ExtSdkDispatch();
    }

    public static ExtSdkDispatch getInstance() {
        return ExtSdkDispatch.SingleHolder.instance;
    }

    @Override
    public void init(@NonNull Object config) {

    }

    @Override
    public void unInit(@Nullable Object params) {

    }

    @Override
    public void addListener(@NonNull ExtSdkListener listener) {

    }

    @Override
    public void delListener(@NonNull ExtSdkListener listener) {

    }

    @Override
    public void callSdkApi(@NonNull String methodType, @Nullable Object params, @Nullable ExtSdkCallback callback) {

    }
}

File: ios/NewClasses/dispatch/ExtSdkDispatch.m
#import "ExtSdkDispatch.h"
#import "ExtSdkDelegateObjc.h"


@interface ExtSdkDispatch () {
    id<ExtSdkDelegateObjc> _listener;
}

@end

@implementation ExtSdkDispatch

+ (nonnull instancetype)getInstance {
    static ExtSdkDispatch *instance = nil;
    static dispatch_once_t predicate;
    dispatch_once(&predicate, ^{
      instance = [[ExtSdkDispatch alloc] init];
    });
    return instance;
}

- (void)addListener:(nonnull id<ExtSdkDelegateObjc>)listener {
    _listener = listener;
}

- (void)callSdkApi:(nonnull NSString *)methodType withParams:(nullable id<NSObject>)params withCallback:(nonnull id<ExtSdkCallbackObjc>)callback {
    
}

- (void)delListener:(nonnull id<ExtSdkDelegateObjc>)listener {
    
}

- (void)init:(nonnull id<NSObject>)config {
    
}

- (void)unInit:(nullable id<NSObject>)params {
    
}

@end
